components/License/helper.ts

```diff
@@ -0,0 +1,65 @@
+import { FeatureAttitude, InfectionRange } from 'license-filter';
+import { i18n } from '../../models/Translation';
+const { t } = i18n;
+
+type OptionValue = Record<string, { value: number; text: string }[]>;
+
+type LicenseTips = Record<string, { text: string }[]>;
+
+const options: string[] = [
+  'popularity',
+  'reuseCondition',
+  'infectionIntensity',
+
+  'jurisdiction',
+  'patentStatement',
+  'patentRetaliation',
+  'enhancedAttribution',
+  'privacyLoophole',
+  'marketingEndorsement',
+];
+
+const optionValue = options.reduce((optionValue: OptionValue, option) => {
+  optionValue[option] = [
+    { value: FeatureAttitude.Undefined, text: t('feature_attitude_undefined') },
+    {
+      value: FeatureAttitude.Positive,
+      text: t('feature_attitude_positive'),
+    },
+    { value: FeatureAttitude.Negative, text: t('feature_attitude_negative') },
+  ];
+  return optionValue;
+}, {});
+
+optionValue.infectionRange = [
+  { value: 0, text: t('infection_range_undefined') },
+  { value: InfectionRange.Library, text: t('infection_range_library') },
+  { value: InfectionRange.File, text: t('infection_range_file') },
+  { value: InfectionRange.Module, text: t('infection_range_module') },
+];
+
+const licenseTips: LicenseTips = {
+  popularity: [
+    {
+      text: t('tip_popularity_0'),
+    },
+    {
+      text: t('tip_popularity_1'),
+    },
+  ],
+  reuseCondition: [{ text: t('tip_reuse_condition') }],
+  infectionIntensity: [{ text: t('tip_infection_intensity') }],
+  jurisdiction: [{ text: t('tip_jurisdiction') }],
+  patentStatement: [{ text: t('tip_patent_statement') }],
+  patentRetaliation: [{ text: t('tip_patent_retaliation') }],
+  enhancedAttribution: [{ text: t('tip_enhanced_attribution') }],
+  privacyLoophole: [{ text: t('tip_privacy_loophole') }],
+  marketingEndorsement: [{ text: t('tip_marketing_endorsement') }],
+  infectionRange: [
+    {
+      text: t('tip_infection_range'),
+    },
+  ],
+};
```

Comment:
```suggestion
const licenseTips: () => LicenseTips = () => ({
  popularity: [
    {
      text: t('tip_popularity_0'),
    },
    {
      text: t('tip_popularity_1'),
    },
  ],
  reuseCondition: [{ text: t('tip_reuse_condition') }],
  infectionIntensity: [{ text: t('tip_infection_intensity') }],
  jurisdiction: [{ text: t('tip_jurisdiction') }],
  patentStatement: [{ text: t('tip_patent_statement') }],
  patentRetaliation: [{ text: t('tip_patent_retaliation') }],
  enhancedAttribution: [{ text: t('tip_enhanced_attribution') }],
  privacyLoophole: [{ text: t('tip_privacy_loophole') }],
  marketingEndorsement: [{ text: t('tip_marketing_endorsement') }],
  infectionRange: [
    {
      text: t('tip_infection_range'),
    },
  ],
});
```
静态数据需要用箭头函数包裹一下，以便 MobX 更新时取到最新的国际化文本。

Reply:
在导入使用时， 不是要实例化？

Reply:
是不是 我要把 生成页面的函数提取到  components/License/helper.ts ？

Reply:
> 是不是 我要把 生成页面的函数提取到  components/License/helper.ts ？ 

和页面啥关系？你按我建议的写就没错。

Reply:
```typescript
{licenseTips[chooseSteps[keyIndex]].map(({ text }) => (
        <p key={text}>{text}</p>
      ))}
```报
在类型 "() => LicenseTips" 上找不到具有类型为 "string" 的参数的索引签名。

Reply:
> > 是不是 我要把 生成页面的函数提取到  components/License/helper.ts ？
> 
> 和页面啥关系？你按我建议的写就没错。

返回一个箭头函数,产生闭包效果 ？

Reply:
> 返回一个箭头函数,产生闭包效果 ？ 

不是返回一个箭头函数，是导出一个箭头函数。

也不是利用什么闭包特性。仅仅是因为模块中的静态数据只在模块解析时生成一遍，要想在组件重绘时更新数据，就不能是纯静态的。所以只有函数才可以在每次组件渲染时重新执行代码，来获取最新的值。